### PR TITLE
docs: Update Vercel deployment documentation.

### PR DESCRIPTION
**Pull request by leerob**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Update the deployment docs for Vercel. Related to https://github.com/nuxt/nuxtjs.org/pull/2004 for v2.

### CODE REVIEW

File: docs/content/3.docs/3.deployment/6.vercel.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ icon: LogoVercel
 
 # Vercel
 
-How to deploy Nuxt to Vercel.
+Nuxt on Vercel supports both static and server-rendered pages.
```
Comment (pi0) [inline docs/content/3.docs/3.deployment/6.vercel.md:7]: At the moment Nuxt 3 / Nitro does is not implementing static generation. Also we support API routes maybe reword like this:

```suggestion
Nuxt on Vercel supports server-rendered pages and API routes.
```

File: docs/content/3.docs/3.deployment/6.vercel.md [hunk 2]
```diff
@@ -13,18 +13,35 @@ How to deploy Nuxt to Vercel.
 - No configuration required
 ::
 
-## Setup
+## CLI
 
-By default, [Vercel](https://vercel.com) will run your `build` script in the `package.json` and automatically detect the built entry point within `.vercel_build_output/`.
+1. Install the [Vercel CLI](https://vercel.com/cli).
+2. Vercel will detect that you are using Nuxt and will enable the correct settings for your deployment.
+3. Your application is deployed! (e.g. [nuxt.vercel.app](https://nuxt.vercel.app/))
```
Comment (pi0) [inline docs/content/3.docs/3.deployment/6.vercel.md:20]: Can we please have access to this deployment? (ideally on vercel.com/nuxt-js)

File: docs/content/3.docs/3.deployment/6.vercel.md [hunk 3]
```diff
@@ -13,18 +13,35 @@ How to deploy Nuxt to Vercel.
...
 
-## Deployment
+```bash
+$ npm i -g vercel
+$ vercel init nuxtjs
```
Comment (pi0) [inline docs/content/3.docs/3.deployment/6.vercel.md:24]:
```suggestion
$ vercel init nuxt
```

File: docs/content/3.docs/3.deployment/6.vercel.md [hunk 4]
```diff
@@ -13,18 +13,35 @@ How to deploy Nuxt to Vercel.
...
+Vercel CLI
+> Success! Initialized "nuxtjs" example in ~/your-folder.
```
Comment (pi0) [inline docs/content/3.docs/3.deployment/6.vercel.md:26]:
```suggestion
> Success! Initialized "nuxt" example in ~/your-folder.
```

File: docs/content/3.docs/3.deployment/6.vercel.md [hunk 5]
```diff
@@ -13,18 +13,35 @@ How to deploy Nuxt to Vercel.
...
+- To deploy, `cd nuxtjs` and run `vercel`.
```
Comment (pi0) [inline docs/content/3.docs/3.deployment/6.vercel.md:27]:
```suggestion
- To deploy, `cd nuxt` and run `vercel`.
```

File: docs/content/3.docs/3.deployment/6.vercel.md [hunk 6]
```diff
@@ -13,18 +13,35 @@ How to deploy Nuxt to Vercel.
...
 
-By default, [Vercel](https://vercel.com) will run your `build` script in the `package.json` and automatically detect the built entry point within `.vercel_build_output/`.
+1. Install the [Vercel CLI](https://vercel.com/cli).
+2. Vercel will detect that you are using Nuxt and will enable the correct settings for your deployment.
+3. Your application is deployed! (e.g. [nuxt.vercel.app](https://nuxt.vercel.app/))
 
-## Deployment
+```bash
+$ npm i -g vercel
+$ vercel init nuxt
```
Comment (pi0) [inline docs/content/3.docs/3.deployment/6.vercel.md:24]: I prefer if we use `nuxi` cli with [vercel starter](https://github.com/nuxt/starter/tree/v3-vercel): `npx nuxi init -t v3-vercel`

File: docs/content/3.docs/3.deployment/6.vercel.md (hunk 2 above)
Comment (leerob) [reply docs/content/3.docs/3.deployment/6.vercel.md:20]: It's coming from here: https://github.com/vercel/vercel/tree/main/examples/nuxtjs

Which is our "single source of truth" for /templates

File: docs/content/3.docs/3.deployment/6.vercel.md (hunk 5 above)
Comment (leerob) [reply docs/content/3.docs/3.deployment/6.vercel.md]: It's actually https://github.com/vercel/vercel/tree/main/examples/nuxtjs here

File: docs/content/3.docs/3.deployment/6.vercel.md (hunk 6 above)
Comment (leerob) [reply docs/content/3.docs/3.deployment/6.vercel.md:24]: `vercel init nuxt` creates a project that uses that same code, from here https://github.com/vercel/vercel/tree/main/examples/nuxtjs

Thoughts?

Comment (pi0) [reply docs/content/3.docs/3.deployment/6.vercel.md:24]: Having an example deployment is nice in vertical but `nuxi init` should be used to make a new nuxt project. Since vercel integration is zero-config, a project made with `nuxi init` should be deployable to vercel via cli as well.

image

File: docs/content/3.docs/3.deployment/6.vercel.md [hunk 7]
```diff
@@ -13,18 +13,35 @@ How to deploy Nuxt to Vercel.
...
+Vercel CLI
+> Success! Initialized "nuxt" example in ~/your-folder.
+- To deploy, `cd nuxt` and run `vercel`.
+```
```
Comment (leerob) [inline docs/content/3.docs/3.deployment/6.vercel.md:28]:
```suggestion
```bash
$ npm i -g vercel
$ npx nuxi init -t v3-vercel
$ vercel
```
```

**Comment by leerob** [comment]
Should be good now 😄

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thank you!
